### PR TITLE
Update documentation for the Google.Cloud.Diagnostics.AspNetCore library

### PR DESCRIPTION
**Pull request by iantalarico**

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/AspNetCoreSnippets.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.Extensions.Logging;
+
+namespace Google.Cloud.Diagnostics.AspNetCore.Snippets
+{
+    public class AspNetCoreSnippets
+    {
+        // Sample: ReportUnandledExceptions
+        public void Configure(IApplicationBuilder app)
+        {
+            // Use first to ensure all unhandled exceptions are reported.
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/AspNetCoreSnippets.cs]: "Use first..." is a little unclear here. Perhaps "Use before handling any requests...", or something else?

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs [hunk 2]
```diff
@@ -23,8 +23,24 @@
...
+    /// {
+    ///     string projectId = "[Google Cloud Platform project ID]";
+    ///     loggerFactory.AddGoogle(projectId);
+    ///     ...
+    /// }
+    /// </code>
+    /// </example>
+    /// 
+    /// <remarks>
+    /// Logs to Google Stackdriver Cloud Logging. See <see cref="GoogleLoggerFactoryExtensions"/>.
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs]: This should probably just be a \<seealso\> under the remarks.

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLoggerFactoryExtensions.cs [hunk 3]
```diff
@@ -22,6 +22,22 @@ namespace Google.Cloud.Diagnostics.AspNetCore
...
+    /// {
+    ///     string projectId = "[Google Cloud Platform project ID]";
+    ///     loggerFactory.AddGoogle(projectId);
+    ///     ...
+    /// }
+    /// </code>
+    /// </example>
+    /// 
+    /// <remarks>
+    /// Logs to Google Stackdriver Cloud Logging. See <see cref="GoogleLogger"/>.
```
Comment (evildour) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLoggerFactoryExtensions.cs]: Same comment here: use \<seealso\> under remarks.

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/AspNetCoreSnippets.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore.Snippets/AspNetCoreSnippets.cs]: Done

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs (hunk 2 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs]: Done

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLoggerFactoryExtensions.cs (hunk 3 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLoggerFactoryExtensions.cs]: Done

**Pull request merged**